exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/handlers/DefaultSqlHandler.java

```diff
@@ -162,6 +166,9 @@ public PhysicalPlan getPlan(SqlNode sqlNode) throws ValidationException, RelConv
 
     log("Optiq Logical", queryRelNode, logger);
     DrillRel drel = convertToDrel(queryRelNode, validatedRowType);
+    if (otherPlan != null) {
```

Comment:
This is a hack. Let's create a relnode which is equivalent to using a pojorecord reader and then when we create the plan, it will effectively be a direct plan.